lib/src/err/mod.rs

```diff
@@ -22,6 +22,10 @@ pub enum Error {
 	#[error("Conditional clause is not truthy")]
 	Ignore,
 
+	/// The database encountered unreachable logic
+	#[error("The database encountered unreachable logic")]
+	Unreachable,
+
```

Comment:
I didn't see where this new error is used, but maybe I scrolled too quickly through similar-looking changes.

Reply:
This error is used in a couple of places. In `options.rs` and `context.rs`. It will be used in more place in due course, to replace some `unreachable!()` uses.